tonic_validate/services/litellm_service.py

```diff
@@ -0,0 +1,131 @@
+            starting_wait_time: float = 1.5,
+            max_retries: int = 12,
+            exp_delay_base: int = 2,
+    ) -> None:
+        """
+        The LiteLLMService class is a wrapper around LiteLLM client for async operations using different LLMs.
+
+        Parameters
+        ----------
+        encoder: Encoding
+            The encoding to use for token count.
+        model: str
+            The model to use for completions.
+        starting_wait_time: float
+            The starting wait time between retries.
+        max_retries: int
+            The maximum number of retries to attempt.
+        exp_delay_base: int
+            Base for exponential back off delay between retries.
+        """
+        try:
+            self.check_environment(model)
+        except Exception as e:
+            logger.error(f"Error: {str(e)}")
+            raise e
+        self.model = model
+        self.encoder = encoder
+        self.max_retries = max_retries
+        self.exp_delay_base = exp_delay_base
+        self.starting_wait_time = starting_wait_time
+        self.cache = LLMCache()
+
+    def check_environment(self, model: str) -> None:
+        if 'gemini' in model:
+            if "GEMINI_API_KEY" not in os.environ:
+                raise Exception(
+                    "GEMINI_API_KEY must be set in the environment when using Gemini"
+                )
+        if 'palm' in model:
+            if "PALM_API_KEY" not in os.environ:
+                raise Exception(
+                    "PALM_API_KEY must be set in the environment when using Palm"
+                )
+        if 'claude' in model:
+            if "ANTHROPIC_API_KEY" not in os.environ:
+                raise Exception(
+                    "ANTHROPIC_API_KEY must be set in the environment when using Claude"
+                )
+
+    async def get_response(self, prompt: str) -> str:
+        """
+        Retrieves a response from the language model asynchronously.
+
+        Parameters
+        ----------
+        prompt: str
+            The prompt to get a response for.
+
+        Returns
+        -------
+        str
+            The response from the language model.
+        """
+
+        async def get_litellm_response():
+            num_retries = 0
+            wait_time = self.starting_wait_time
+            while num_retries < self.max_retries:
+                try:
+                    messages = [
+                        {"role": "system",
+                         "content": "You are a helpful assistant. Respond using markdown.",
+                         },
+                        {"role": "user", "content": prompt},
+                    ]
+                    response = await acompletion(
+                        model=self.model,
+                        messages=messages,
+                        temperature=0.0,
+                    )
+                    response_content = response.choices[0].message.content
```

Comment:
I think the typing is messed up here since it seems litellm can return multiple types. The type checker lists this as invalid even though the code works fine